scompose/project/project.py

```diff
@@ -223,8 +230,9 @@ def _sort_instances(self, instances):
                 index = sorted_instances.index(instance)
 
             for dep in depends_on:
-                if not dep in sorted_instances:
-                    sorted_instances.insert(index, dep)
+                for inst in instances.values():
+                    if dep == inst.name:
```

Comment:
So here we are using just name and not get_replica_name because any of the replicas satisfies the condition (e.g., we insert them in the same block?

Reply:
And I remember correctly, the depends_on wouldn't include the replica number.

Reply:
You got the rationale right. We use the name (and not the get_replica_name) because when we write the dependencies in the compose.yml file, we don't include the replica number. 

The existing sorting method will place all dependent replicas before the 'dependant' instance: 

For instance:
hello ( replicas: 3) depends on hallo ( replicas: 1). After `self._sort_instances(self.instances)` it will be: `hello1`,  `hello2`, `hello3` (order doesn't matter for replicas) and then `hallo1).

Reply:
Ooops, I wrote my example in the wrong way.. let me fix it before it becomes confusing